coralproject/talk
# Hook up remaining DSA notifications

## What does this PR do?

- Hooks up the DSA notifications that happen when a user's comment is deemed illegal
- Hooks up DSA reason information when comments are rejected into notifications

## These changes will impact:


## What changes to the GraphQL/Database Schema does this PR introduce?

None

## Does this PR introduce any new environment variables or feature flags?

No

## If any indexes were added, were they added to `INDEXES.md`?

No new indices.

## How do I test this PR?

DSA Reporting:
- Create a comment
- Report it with the DSA reporting
- Process the comment with an admin/mod user in the DSA report flow
- Decide the comment is illegal in the DSA report flow
    - Specify the grounds and explanation
- Check the original user's notifications
    - See a new notification showing the illegal reasoning, grounds, explanation, and the referenced comment
- Check the reporting user's notifications
    - See a new notification indicating that the report decision was made

General Approve/Reject of comments with reasons:
- Create a comment
- Reject it using an admin/moderator account in the mod queues
    - Provide a reason
- Check notifications of the original commenter
    - See that they have a notification indicating the rejection code and other legal details

## Where any tests migrated to React Testing Library?

No

## How do we deploy this PR?

No special considerations, it goes into the DSA epic branch when complete.
